packages/core/src/blks/BulletListItem/definition.ts

```diff
@@ -0,0 +1,100 @@
+import { updateBlockTr } from "../../api/blockManipulation/commands/updateBlock/updateBlock.js";
+import { getBlockInfoFromTransaction } from "../../api/getBlockInfoFromPos.js";
+import { defaultProps } from "../../blocks/defaultProps.js";
+import { getListItemContent } from "../../blocks/ListItemBlockContent/getListItemContent.js";
+import {
+  createBlockConfig,
+  createBlockNoteExtension,
+  createBlockSpec,
+} from "../../schema/blocks/playground.js";
+import { handleEnter } from "../utils/listItemEnterHandler.js";
+
+const config = createBlockConfig(() => ({
+  type: "bulletListItem" as const,
+  propSchema: {
+    ...defaultProps,
+  },
+  content: "inline",
+}));
+
+export const definition = createBlockSpec(config).implementation(
+  () => ({
+    parse(element) {
+      if (element.tagName !== "LI") {
+        return false;
+      }
+
+      const parent = element.parentElement;
+
+      if (parent === null) {
+        return false;
+      }
+
+      if (
+        parent.tagName === "UL" ||
+        (parent.tagName === "DIV" && parent.parentElement?.tagName === "UL")
+      ) {
+        return {};
+      }
+
+      return false;
+    },
+    // As `li` elements can contain multiple paragraphs, we need to merge their contents
+    // into a single one so that ProseMirror can parse everything correctly.
+    parseContent: ({ el, schema }) =>
+      getListItemContent(el, schema, "bulletListItem"),
+    render() {
+      const div = document.createElement("div");
+      // We use a <p> tag, because for <li> tags we'd need a <ul> element to put
+      // them in to be semantically correct, which we can't have due to the
+      // schema.
+      const el = document.createElement("p");
+
+      div.appendChild(el);
+
+      return {
+        dom: div,
+        contentDOM: el,
+      };
+    },
+  }),
+  () => [
+    createBlockNoteExtension({
```

Comment:
What was the reasoning again to define keyboardshortcuts / inputrules in a separate argument + `createBlockNoteExtension`?

Reply:
This introduces keyboard shortcuts & inputrules to the extension API because we didn't have a way of doing them before.

I considered exposing them to both APIs (e.g. a custom block has an input rule API & so does the extension API), but that only added complexity.

Allowing extensions to be included to the block API is useful in that some things may require an extension to function (e.g. code block syntax highlighting, numbered list decorations).